conans/test/unittests/tools/files/test_downloads.py

```diff
@@ -157,6 +158,33 @@ def test_download_no_retries_errors(self, bottle_server):
         assert "Waiting" not in str(conanfile.output)
         assert "retry" not in str(conanfile.output)
 
+    def test_download_localfile(self):
+        conanfile = ConanFileMock()
+        conanfile._conan_requester = requests
+
+        file_location =  os.path.join(temp_folder(), "file.txt")
+        with open(file_location, 'w') as textfile:
+            textfile.write('this is some content\n')
+
+        file_url = f"file:///{file_location}"
+        file_md5 = "a0b156435474e688206c68e5c66a3327"
+        
+        dest = os.path.join(temp_folder(), "downloaded_file.txt")
+        download(conanfile, file_url, dest, md5=file_md5)
+        content = load(dest)
+        assert "this is some content" == content.rstrip()
```

Comment:
```suggestion
        assert "this is some content" == content
```

The rstrip() is probably unnecessary, and risks hiding a bug somewhere. The file must be identical

Reply:
Fixed it by changing the md5 that is checked above (as it stands, given the md5 check, this is probably redundant).

Issue was that most command line text editors and the shell itself will append a newline character, so all of my attempts to get the md5 sum of a string will give me the md5 sum of a string ending with the newline character. Given that a few lines above we have control of the exact byte-wise contents of the file, I think it's safe to remove the `rstrip()`